src/utils/apiRequests/api.ts

```diff
@@ -312,25 +348,33 @@ export async function putAuthenticatedRequest<T>(
   url: any,
   data: any,
   token: any,
-  impersonatedEmail?: string,
   errorHandler?: Function
 ): Promise<T | ApiCustomError | undefined> {
+  const targetData = localStorage.getItem("impersonationData") ;
+  const impersonatedDataFromLocal = JSON.parse(targetData);
+  const header : any = {
+    'Content-Type': 'application/json',
+    'tenant-key': `${TENANT_ID}`,
+    'X-SESSION-ID': await getsessionId(),
+    Authorization: `Bearer ${token}`,
+    'x-locale': `${
+      localStorage.getItem('language')
+        ? localStorage.getItem('language')
+        : 'en'
+    }`,
+  }
+  if(impersonatedDataFromLocal?.targetEmail ){
+    header["X-SWITCH-USER"] =  impersonatedDataFromLocal?.targetEmail 
+  }
+
+  if( impersonatedDataFromLocal?.supportPin){
+    header["X-USER-SUPPORT-PIN"] =  impersonatedDataFromLocal?.supportPin 
+  }
```

Comment:
You could abstract this logic to a common function as it used in all authenticated requests.